assets/js/layouts/playground.js

```diff
@@ -26,11 +30,21 @@ export class Playground {
 		return doc;
 	}
 
-	async switchTo(link) {
+	preload(target) {
+		const newUrl = this.image.currentSrc.replace(
+			this.wrapper.dataset.image,
+			target.dataset.image
+		);
+
+		new Image().src = newUrl;
```

Comment:
I think it's clearer because when reading the code, you can separate the lines into "one builds the URL and the other preloads". If both are combined, it's harder to follow what's happening, especially because the URL logic is already complex.